src/pages-helpers/pupil/options-pages/options-pages-helpers.ts

```diff
@@ -8,52 +9,37 @@ import { PupilProgrammeListingData } from "@/node-lib/curriculum-api-2023/querie
 import OakError from "@/errors/OakError";
 import curriculumApi2023 from "@/node-lib/curriculum-api-2023";
 import { resolveOakHref } from "@/common-lib/urls";
+import { getAvailableProgrammeFactor } from "@/components/PupilViews/PupilProgrammeListing/getAvailableProgrammeFactor";
+import { ExamboardData } from "@/components/PupilComponents/BrowseExamboardSelector";
+import { TierData } from "@/components/PupilComponents/BrowseTierSelector";
+import { PupilViewsProgrammeListingProps } from "@/components/PupilViews/PupilProgrammeListing/PupilProgrammeListing.view";
 
 export type OptionsURLParams = {
   programmeSlug: string;
+  examboardSlug?: string;
 };
 
-export type ProgrammesPageProps = {
-  baseSlug: string;
-  programmes: PupilProgrammeListingData[];
-  yearSlug: PupilProgrammeListingData["yearSlug"];
-  examboardSlug?: ProgrammeFields["examboard_slug"] | null;
-};
-
-export const getYearSlug = ({
-  programmes,
-}: {
-  programmes: PupilProgrammeListingData[];
-}) => {
-  const yearSlug = programmes[0]?.yearSlug;
-
-  if (
-    programmes.filter((programme) => programme.yearSlug !== yearSlug).length >
-      0 ||
-    !yearSlug
-  ) {
-    throw new OakError({ code: "curriculum-api/params-incorrect" });
-  }
-  return yearSlug;
-};
-
-export const isExamboardSlug = (
-  examboardSlug: ProgrammeFields["examboard_slug"] | string | null,
-): examboardSlug is ProgrammeFields["examboard_slug"] =>
-  Object.keys(examboardSlugs.Values).includes(examboardSlug ?? "");
-
 export const getPupilOptionData = async (
   context: GetStaticPropsContext<OptionsURLParams>,
-): Promise<GetStaticPropsResult<ProgrammesPageProps>> => {
-  if (!context.params) {
+): Promise<GetStaticPropsResult<PupilViewsProgrammeListingProps>> => {
+  if (!context.params || !context.params.programmeSlug) {
     throw new OakError({ code: "curriculum-api/params-incorrect" });
   }
 
-  // For the options path rename programmeSlug to baseSlug as this is the accurate usage of the options page.
+  // For the options route we rename programmeSlug to baseSlug as this is the accurate usage of the options page.
   // I would have created a new folder [baseSlug] but multiple dynamic params on the same segment is not allowed.
-  const { programmeSlug: baseSlug } = context.params;
+  const { programmeSlug: baseSlug, examboardSlug: rawExamboardSlug = null } =
+    context.params;
 
-  if (!baseSlug) {
+  let examboardSlug: ProgrammeFields["examboard_slug"] | null = null;
+  if (rawExamboardSlug) {
+    if (!isExamboardSlug(rawExamboardSlug)) {
+      throw new OakError({ code: "curriculum-api/params-incorrect" });
+    }
+    examboardSlug = rawExamboardSlug;
+  }
+
+  if (examboardSlug && !isExamboardSlug(examboardSlug)) {
```

Comment:
this looks like it could be simplified ?
`const examboardSlug: rawExamboardSlug | ProgrammeFields["examboard_slug"] | null = null;
 if (examboardSlug && !isExamboardSlug(examboardSlug)) {
    throw new OakError({ code: "curriculum-api/params-incorrect" });
  }
`

Reply:
Yeah, I don't think most of this is needed, will push update.